extensions/lifecycle/conductor/LifecycleConductor.js

```diff
@@ -467,18 +498,30 @@ class LifecycleConductor {
                     return cb(err);
                 }
 
-                const batch = buckets.map(bucket => {
-                    const [canonicalId, bucketUID, bucketName] =
-                              bucket.split(':');
-                    if (!canonicalId || !bucketUID || !bucketName) {
-                        log.error(
-                            'malformed zookeeper bucket entry, skipping',
-                            { zkPath: zkBucketsPath, bucket });
-                        return null;
-                    }
+                const batch = buckets
+                    .filter(bucket => {
+                        const [canonicalId, bucketUID, bucketName] =
+                                bucket.split(':');
+                        if (!canonicalId || !bucketUID || !bucketName) {
+                            log.error(
+                                'malformed zookeeper bucket entry, skipping',
+                                { zkPath: zkBucketsPath, bucket });
+                            return false;
+                        }
 
-                    return { canonicalId, bucketName };
-                });
+                        if (this._isBlacklisted(canonicalId, bucketName)) {
+                            return false;
+                        }
+
+                        return true;
+                    })
+                    .map(bucket => {
+                        const split = bucket.split(':');
+                        const canonicalId = split[0];
+                        const bucketName = split[2];
```

Comment:
Shouldnt we check for length before accessing `[2]` ?

Reply:
Already filtering out invalid entries in the .filter() above